src/box/iproto.cc

```diff
@@ -3377,6 +3510,94 @@ iproto_thread_init(struct iproto_thread *iproto_thread)
 	rlist_create(&iproto_thread->connections);
 }
 
+/**
+ * Returns `enum iproto_type' if `name' is a valid name of the IPROTO overriding
+ * event, or iproto_type_MAX otherwise. If the name contains correct prefix, but
+ * the request type is invalid, the error is logged with CRIT log level.
+ * `is_by_id' set to True if the request is overridden by id, False if by name.
+ */
+static uint32_t
+get_iproto_type_from_event_name(const char *name, bool *is_by_id)
+{
+	*is_by_id = false;
+	const char *prefix = "box.iproto.override";
+	const size_t prefix_len = strlen(prefix);
+	if (strncmp(name, prefix, prefix_len) != 0)
+		return iproto_type_MAX;
+
+	const char *req_name = name + prefix_len;
+	uint32_t type = iproto_type_MAX;
+
+	if (*req_name == '.') {
+		/* Skip the dot. */
+		req_name++;
+		type = get_iproto_type_by_name(req_name);
+	} else if (*req_name == '[') {
+		/* Skip open bracket. */
+		req_name++;
+		if (isdigit(*req_name) || *req_name == '-') {
+			*is_by_id = true;
+			char *endptr;
+			type = strtol(req_name, &endptr, 10);
+			if (endptr == req_name) {
+				type = iproto_type_MAX;
+			} else {
+				/*
+				 * At least one digit is parsed.
+				 * Check that the rest of the string equals "]".
+				 */
+				if (*endptr != ']' || endptr[1] != 0)
+					type = iproto_type_MAX;
+			}
+		}
+	}
+
+	if (type == iproto_type_MAX) {
+		say_crit("The event `%s' is in IPROTO override namespace, but "
+			 "`%s' is not a valid request type", name, req_name);
```

Comment:
It's not correct to use `req_name` here since it was advanced during the name parsing. Let's use `name + prefix_name`.

Reply:
Changed to:
```diff
@@ -3650,11 +3654,13 @@ get_iproto_type_from_event_name(const char *name, bool *is_by_id)
                return iproto_type_MAX;

        const char *req_name = name + prefix_len;
+       const char *req_name_err = req_name;
        uint32_t type = iproto_type_MAX;

        if (*req_name == '.') {
                /* Skip the dot. */
                req_name++;
+               req_name_err = req_name;
                type = get_iproto_type_by_name(req_name);
        } else if (*req_name == '[') {
                /* Skip open bracket. */
@@ -3674,11 +3680,14 @@ get_iproto_type_from_event_name(const char *name, bool *is_by_id)
                                        type = iproto_type_MAX;
                        }
                }
+       } else {
+               return iproto_type_MAX;
        }

        if (type == iproto_type_MAX) {
                say_crit("The event `%s' is in IPROTO override namespace, but "
-                        "`%s' is not a valid request type", name, req_name);
+                        "`%s' is not a valid request type",
+                        name, req_name_err);
                return iproto_type_MAX;
        }
        bool is_unsupported_request = type == IPROTO_JOIN ||
```